library/entity/points_of_interest/src/main/java/org/quiltmc/qsl/points_of_interest/api/PointOfInterestHelper.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.points_of_interest.api;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+import com.google.common.collect.ImmutableSet;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.poi.PointOfInterest;
+import net.minecraft.world.poi.PointOfInterestType;
+import org.quiltmc.qsl.points_of_interest.mixin.PointOfInterestTypeAccessor;
+
+/**
+ * This class provides utilities to create a {@link PointOfInterestType}.
+ *
+ * <p>A point of interest is typically used by villagers to specify their workstation blocks, meeting zones and homes.
+ * Points of interest are also used by bees to specify where their bee hive is and nether portals to find existing portals.
+ */
+public final class PointOfInterestHelper {
+
+	/**
+	 * Creates and registers a {@link PointOfInterestType}.
+	 *
+	 * @param id The id of this {@link PointOfInterestType}.
+	 * @param ticketCount the amount of tickets.
```

Comment:
I went ahead and added more info to the javadocs, but I'll also comment here just in case.

The amount of tickets is how many things can focus on this POI at once. Most of them have a default of 1. This is what allows only one villager to use a given job block, but many villagers to head to meeting points.

Search distance is just how close to the POI an entity has to be in blocks before considering it reached.